### PR TITLE
[EPT-1713] Add modal to delete page

### PR DESCRIPTION
**Pull request by shanthru**
[EPT-1713](https://sailthru.atlassian.net/browse/EPT-1713)

This PR adds a confirmation to delete a modal and a notifier to display success and failure notices.

For working functionality, please use this PHP branch:
https://github.com/sailthru/php/pull/5229

[EPT-1713]: https://sailthru.atlassian.net/browse/EPT-1713?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webpack/webpack.build.js [hunk 1]
```diff
@@ -32,6 +32,13 @@ module.exports = function ({ basePath }) {
           test: /\.(png|jpg|gif|svg)(\?[\s\S]+)?$/,
           type: "asset/resource",
         },
+        {
+          test: /\.css$/,
+          use: [
+            'style-loader',
+            'css-loader'
+          ]
+        },
```
Comment (shanthru) [inline webpack/webpack.build.js:41]: Followed the below guides for these settings:

https://webpack.js.org/loaders/css-loader/#pure-css-css-modules-and-postcss
https://blog.jakoblind.no/css-modules-webpack/

File: src/ListView/DeletePageModal.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    >
+      <p>
+        Page <strong>{pageName}</strong>
+      </p>
+      <p>Are you sure you want to delete this page?</p>
+    </Modal>
+  );
+}
+
+export { DeletePageModal };
```
Comment (lindsaycooper) [inline src/ListView/DeletePageModal.jsx:52]: Shouldn't this follow the pattern from `CreatePage` in terms of directory structure? Are you reconsidering this approach?
image

File: src/ListView/DeletePageModal.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  function handleSubmit() {
+    setDisableSubmitButton(true);
+    deletePage(pageId)
+      .then(function () {
+        setNotificationType("success");
+        setPageDeletedMessage(PageSuccessDeleteMessage);
+        return true;
+      })
+      .catch(function (error) {
+        console.log("Error deleting page", error);
```
Comment (lindsaycooper) [inline src/ListView/DeletePageModal.jsx:27]:
```suggestion
        console.error("Error deleting page", pageId, error);
```

File: src/shared/ResultNotification.jsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    return null;
+  }
+  return (
+    <Notification className={styles.notifier} type={type}>
+      {message}
+    </Notification>
+  );
+}
+
+export { ResultNotification };
```
Comment (lindsaycooper) [inline src/shared/ResultNotification.jsx:16]: question: What is the difference between the `shared` and `core` directories?

File: src/ListView/ListView.jsx [hunk 5]
```diff
@@ -41,7 +48,11 @@ function ListView() {
         }, {});
         setPageMap(map);
       })
-      .catch(console.error)
+      .catch(function (error) {
+        console.log("Error loading pages", error);
```
Comment (lindsaycooper) [inline src/ListView/ListView.jsx:52]:
```suggestion
        console.error("Error fetching pages", error);
```

File: src/ListView/ListView.jsx [hunk 6]
```diff
@@ -65,10 +76,17 @@ function ListView() {
           </Button>
         }
       >
+        <ResultNotification
+          display={notificationType}
+          message={notificationMessage}
+          type={notificationType}
```
Comment (lindsaycooper) [inline src/ListView/ListView.jsx:82]: question: Why is `notificationType` used for both `display` and `type`?

File: src/ListView/PageTableHoverOptions.jsx [hunk 7]
```diff
@@ -1,32 +1,54 @@
...
     {
       label: "Delete",
       icon: "fal fa-trash fa-fw",
       iconHover: "fas fa-trash fa-fw",
       onClick: () => {
-        console.log("Deleting", pageId);
-        //         displayModal({
-        // TODO see EPT-1713
-        //         });
+        setDisplayDeleteModal((state) => !state);
```
Comment (lindsaycooper) [inline src/ListView/PageTableHoverOptions.jsx:30]: nit-pick: Can this `onClick` be one line like "Edit"?

Comment (lindsaycooper) [reply src/ListView/PageTableHoverOptions.jsx:30]: question: Is there a reason this cannot be simplified to this?
```suggestion
        setDisplayDeleteModal(true);
```

File: src/ListView/DeletePageModal.jsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        setNotificationType("error");
+        setPageDeletedMessage(PageErrorDeleteMessage);
+      })
+      .finally(function () {
+        window.location.href = "/hosted-pages-list";
```
Comment (lindsaycooper) [inline src/ListView/DeletePageModal.jsx:32]: question: Does this do a page reload?

File: src/shared/ResultNotification.jsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { Notification } from "@sailthru/stui-components";
+import styles from "./styles.module.css";
+
+function ResultNotification({ display, message, type }) {
+  if (!display) {
+    return null;
+  }
```
Comment (lindsaycooper) [inline src/shared/ResultNotification.jsx:8]: If you aren't explicitly setting display, I'd suggest removing it and using the other 2 variables for this check
```suggestion
  if (!type || !message) {
    return null;
  }
```

File: src/ListView/ListView.jsx [hunk 10]
```diff
@@ -6,10 +6,17 @@ import * as pagesApi from "../core/pagesApi";
...
+import { ResultNotification } from "../shared/ResultNotification";
+
+const LoadPagesErrorMessage =
+  "Something went wrong and hosted pages could not be retrieved. Please refresh and try again.";
 
 function ListView() {
   const [loading, setLoading] = useState(true); // load from the start to prevent blip
   const [pageMap, setPageMap] = useState({});
+  const [notificationType, setNotificationType] = useState("");
+  const [notificationMessage, setNotificationMessage] = useState("");
```
Comment (lindsaycooper) [inline src/ListView/ListView.jsx:18]: question: Once these are set (due to an error), how are they reset? How does the notification disappear?

File: src/ListView/DeletePageModal.jsx (hunk 8 above)
Comment (shanthru) [reply src/ListView/DeletePageModal.jsx:32]: Yup

File: src/ListView/ListView.jsx (hunk 10 above)
Comment (shanthru) [reply src/ListView/ListView.jsx:18]: For notifications from the DeletePageModal, they disappear after the page reloads
https://github.com/sailthru/hosted-pages-frontend/blob/586fc779bf0882a21360b0702c3fbf0798ddc4c3/src/ListView/DeletePageModal.jsx#L32

For the ListView, if there is an error loading pages, the message does not disappear but requests the user to refresh the page. This is consistent with the templates-list page.

File: src/ListView/ListView.jsx (hunk 6 above)
Comment (shanthru) [reply src/ListView/ListView.jsx]: I've updated this with a new var specifically for display as this did seem confusing
https://github.com/sailthru/hosted-pages-frontend/pull/13/commits/cce2d912a6d38dee3754b27a1af99ac052c727bb

File: src/ListView/PageTableHoverOptions.jsx (hunk 7 above)
Comment (shanthru) [reply src/ListView/PageTableHoverOptions.jsx:31]: I kept this just as following the example in STUI
https://stui.design.sailthru.com/?path=/docs/components_design-system-components-modal--as-default

File: src/shared/ResultNotification.jsx (hunk 9 above)
Comment (shanthru) [reply src/shared/ResultNotification.jsx:8]: I've updated this to specifically set display

File: src/ListView/ListView.jsx (hunk 5 above)
Comment (shanthru) [reply src/ListView/ListView.jsx]: updated

File: src/ListView/DeletePageModal.jsx (hunk 2 above)
Comment (shanthru) [reply src/ListView/DeletePageModal.jsx]: Updated the directory structure per our chat.

File: src/shared/ResultNotification.jsx (hunk 4 above)
Comment (shanthru) [reply src/shared/ResultNotification.jsx]: Updated the directory structure per our chat.

File: src/ListView/PageTableHoverOptions.jsx (hunk 7 above)
Comment (lindsaycooper) [reply src/ListView/PageTableHoverOptions.jsx:31]: non-blocking: The `onDismiss` is using the simplified approach. I find the negating state more confusing than just `true`. But also not a major issue either.
```
onDismiss={() => setDisplayDeleteModal(false)}
```

File: src/ListView/DeletePageModal.jsx [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+const PageSuccessDeleteMessage = "Page successfully deleted.";
+const PageErrorDeleteMessage =
+  "Page was unable to be deleted, please try again later.";
+
+function DeletePageModal({
+  pageId,
+  pageName,
+  setDisplayDeleteModal,
+  setDisplayNotification,
```
Comment (lindsaycooper) [inline src/ListView/DeletePageModal.jsx:13]: nit-pick: The other notification variables follow the pattern `setNotification<Variable>`
```suggestion
  setNotificationDisplay,
```

**Review by lindsaycooper — APPROVED** [review]

**Review by lindsaycooper — APPROVED** [review]
### QA +1

- [x] Wireframe and feature look similar

Hover dropdown
- [x] Edit → Takes you to HTML page
- [x] Delete → Show modal with name of page

Delete Modal
- [x] Delete → Deletes page (on list page and database) → Modal closes → List page refreshes → Success toast appears for 10 sec.
- [x] Cancel → Goes back to list page
- [x] Clicking X → Goes back to list page

Table
- [x] Page Name → Takes you to HTML editor
- [x] URL → Takes you to URL in new tab

Error state
- [x] Mock network request on delete failed → Error toast appears ~~for 10 sec.~~
- Notes
  - Toast doesn't disappear after 10 seconds (this behavior matches the template list page)
  - ❓ I think the error message could be improved

Current Error Message:
```
Something went wrong and hosted pages could not be retrieved. 
Please refresh and try again.
```
My suggestion: 
```
Something went wrong loading the hosted pages. 
Please refresh the page to try again.
```

File: src/ListView/PageTableHoverOptions.jsx (hunk 7 above)
Comment (shanthru) [reply src/ListView/PageTableHoverOptions.jsx]: updated this to true but the new line is a `prettier` change

**Review by lindsaycooper — APPROVED** [review]

**Pull request merged**